sigh/lib/sigh/runner.rb

```diff
@@ -81,13 +81,6 @@ def fetch_profiles
 
       # Take the provisioning profile name into account
       results = filter_profiles_by_name(results) if Sigh.config[:provisioning_name].to_s.length > 0
-
-      # Since September 20, 2016 spaceship doesn't distinguish between AdHoc and AppStore profiles
-      # any more, since it requires an additional request
-      # Instead we only call is_adhoc? on the matching profiles to speed up spaceship
-
-      results = filter_profiles_for_adhoc_or_app_store(results)
```

Comment:
Are we sure we need to remove this method?
If I call sigh to download adhoc provision profile with a specific name, and in the dev portal I only have an appstore provision profile with that name, the filtering will prevent this method from returning the profile I did not requested, I think?

Reply:
Hi @csknns,
Previously, yes the method `filter_profiles_for_adhoc_or_app_store()` had the responsibility of returning the right profile between AdHoc and AppStore.
Because AdHoc was treated as AppStore (adhoc profiles objects were instances of the class AppStore same as for appstore profiles) and we needed to read `is_adhoc?` (does the profile have devices in it or not) to make the distinction between them.

Now that is no more the case, `all()` method in `Spaceship::Portal::ProvisioningProfile` is doing the job correctly as it relies directly on the type.

Reply:
@rkrim Yep you are right, now filtering by class type works as expected 👍 . This PR seems to better address the issue so I am closing my pull request.